lib/src/iterable_extensions.dart

```diff
@@ -22,6 +22,15 @@ extension IterableExtensions<E> on Iterable<E> {
     forEach(fn);
     return this;
   }
+
+  /// Returns a lazily filtered [Iterable] (via [Iterable.where]) using the
+  /// provided [predicate] when the supplied [condition] is `true`. Otherwise
+  /// returns the original, un-filtered, [Iterable].
+  ///
+  /// `since 0.2.0`
+  Iterable<E> filterWhen(bool condition, bool Function(E element) predicate) {
+    return condition ? where(predicate) : this;
```

Comment:
I considered calling it `whereWhen`... But that doesn't sound right when reading. `filterWhen` IMO was a better term. 

I checked [darq](https://pub.dev/packages/darq) for a similar API. Found nothing.

Reply:
could just use a predicate which is `condition || predicate` . But I have no problem adding this if you think it is useful.

Reply:
I considered that initially, but the condition is checked for each item. That can be very expensive depending on the size of the iterable.